sled-hardware/src/illumos/mod.rs

```diff
@@ -534,11 +535,22 @@ impl HardwareManager {
         // receiver will receive a tokio::sync::broadcast::error::RecvError::Lagged
         // error, indicating they should re-scan the hardware themselves.
         let (tx, _) = broadcast::channel(1024);
-        let hw = match stub_scrimlet {
-            None => HardwareView::new().map_err(|e| e.to_string())?,
-            Some(active) => HardwareView::new_stub_tofino(active)
-                .map_err(|e| e.to_string())?,
-        };
+        let hw = match scrimlet_override {
+            None => HardwareView::new(),
+            Some(mode) => match mode {
+                ScrimletMode::Disabled => HardwareView::new_stub_tofino(
+                    // active=
+                    false,
+                ),
+                ScrimletMode::Stub => HardwareView::new_stub_tofino(true),
+                // TODO: correctness
+                // I'm not sure whether or not we should be treating softnpu
+                // as a stub or treating it as a real ASIC here, so this
+                // might change.
+                ScrimletMode::Softnpu => HardwareView::new_stub_tofino(true),
```

Comment:
This refers to how we look for the driver and devices files. This can't be a "real" Tofino, though we may want to add a variant to `TofinoView` for the SoftNPU-specific stuff.